app/javascript/forms/provider-forms/cloud-provider-form/index.jsx

```diff
@@ -0,0 +1,170 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { componentTypes, validatorTypes } from '@data-driven-forms/react-form-renderer';
+
+import { API } from '../../../http_api';
+import MiqFormRenderer from '../../data-driven-form';
+import miqRedirectBack from '../../../helpers/miq-redirect-back';
+
+const typeSelectorSchema = {
+  fields: [
+    {
+      component: componentTypes.SELECT,
+      name: 'type',
+      label: __('Type'),
+      placeholder: `<${__('Choose')}>`,
+      loadOptions: () =>
+        API.options('/api/providers').then(({ data: { supported_providers } }) => supported_providers // eslint-disable-line camelcase
+          .filter(({ kind }) => kind === 'cloud')
+          .map(({ title, type }) => ({ value: type, label: title }))),
+    },
+  ],
+};
+
+const loadProviderServerZones = () =>
+  API.get('/api/zones?expand=resources&attributes=id,name,visible&filter[]=visible!=false&sort_by=name')
+    .then(({ resources }) => resources.map(({ name }) => ({ value: name, label: name })));
```

Comment:
Oh, this bit sounds wrong,
should we really be selecting zones based on name not id?

Reply:
Apparently that is what API wants. There was ID before.

Reply:
Well, it IS wrong, but this is what the API wants :laughing:

Reply:
Ah :D :+1:

Reply:
Though, in that case, we probably don't need `attributes=id,visible`